web/core/src/jsMain/kotlin/org/jetbrains/compose/web/elements/RadioGroup.kt

```diff
@@ -0,0 +1,120 @@
+
+/**
+ * @param [value] - sets `value` attribute
+ * @param [id] - sets `id` attribute
+ * @param [attrs] - builder to set any eligible attribute
+ */
+@Composable
+@NonRestartableComposable
+@ExperimentalComposeWebApi
+fun <T> RadioGroupScope<T>.RadioInput(
+    value: T,
+    id: String? = null,
+    attrs: (InputAttrsBuilder<Boolean>.() -> Unit)? = null
+) {
+    val checkedValue = getCompositionLocalRadioGroupCheckedValue()
+    val radioGroupName = getCompositionLocalRadioGroupName()
+
+    Input(
+        type = InputType.Radio,
+        attrs = {
+            attrs?.invoke(this)
+            if (id != null) id(id)
+            name(radioGroupName)
+
+            val valueString = value.toString()
+            checked(checkedValue == valueString)
+            value(valueString)
+        }
+    )
+}
+
+/**
+ * @param [checkedValue] - value of a radio input that has to be checked
+ * @param [name] - radio group name. It has to be unique among all radio groups.
+ * If it's null during first composition, radio group will use a generated name.
+ * @param [content] - is a composable lambda that contains any number of [RadioInput]
+ */
+@Composable
+@NonRestartableComposable
+@ExperimentalComposeWebApi
+fun <E, T : Enum<E>?> RadioGroup(
+    checkedValue: T,
+    name: String? = null,
+    content: @Composable RadioGroupScope<T>.() -> Unit
+) {
+    val radioGroupName = remember { name ?: generateNextRadioGroupName() }
+
+    CompositionLocalProvider(
+        radioGroupCompositionLocalValue provides checkedValue.toString(),
+        radioGroupCompositionLocalName provides radioGroupName,
+        content = {
+            // normal cast would fail here!
+            // this is to specify the type of the values for radio inputs
+            content(RadioGroupScopeImpl.unsafeCast<RadioGroupScope<T>>())
+        }
+    )
+}
+
+/**
+ * @param [checkedValue] - value of a radio input that has to be checked
+ * @param [name] - radio group name. It has to be unique among all radio groups.
+ * If it's null during first composition, radio group will use a generated name.
+ * @param [content] - is a composable lambda that contains any number of [RadioInput]
+ */
+@Composable
+@NonRestartableComposable
+@ExperimentalComposeWebApi
+fun RadioGroup(
+    checkedValue: String?,
+    name: String? = null,
+    content: @Composable RadioGroupScope<String>.() -> Unit
+) {
+    val radioGroupName = remember { name ?: generateNextRadioGroupName() }
+
+    CompositionLocalProvider(
+        radioGroupCompositionLocalValue provides checkedValue,
+        radioGroupCompositionLocalName provides radioGroupName,
+        content = {
+            // normal cast would fail here!
+            // this is to specify the type of the values for radio inputs
+            content(RadioGroupScopeImpl.unsafeCast<RadioGroupScope<String>>())
+        }
+    )
+}
+
+@ExperimentalComposeWebApi
+open class RadioGroupScope<T> internal constructor()
```

Comment:
`RadioGroupScope` helps to infer the type of values for Radio Buttons within a particular RadioGroup:

```kotlin
RadioGroup<MyValues> { // RadioGroupScope<MyValues>

    RadioInput(value = MyValues.Value1)
    RadioInput(value = MyValues.Value2)
    RadioInput(value = SomeOtherValues.Value1) // this won't compile
}
```

we need to have it public since it's used in the scope of RadioGroup content. 

As for being it open, I think it's better to change it to:

```kotlin
@ExperimentalComposeWebApi
class RadioGroupScope<T> internal constructor()

@OptIn(ExperimentalComposeWebApi::class)
private val radioGroupScopeImpl = RadioGroupScope<Any>()
```

`radioGroupScopeImpl`  helps to reuse the same instance for different radio groups with different values types. This relies on usafeCast<>() though. I think the advantage of reusing the same instance  (instead of creating a new object every time) surpasses the cons of unsafeCast.